typescript/src/lib/ethereum/wallet-registry.ts

```diff
@@ -66,13 +66,16 @@ export class EthereumWalletRegistry
    * @see {WalletRegistry#getWalletPublicKey}
    */
   async getWalletPublicKey(walletID: Hex): Promise<Hex> {
-    const publicKey = await backoffRetrier<string>(this._totalRetryAttempts)(
-      async () => {
-        return await this._instance.getWalletPublicKey(
-          walletID.toPrefixedString()
-        )
-      }
-    )
+    const publicKey = await backoffRetrier<string>(
+      this._totalRetryAttempts,
+      undefined,
+      undefined,
+      skipRetryWhenMatched(["Wallet with the given ID has not been registered"])
```

Comment:
I'm not sure we can safely skip this error. We want to work around the problem in the redemption process, but the `getWalletPublicKey` function can be used in other processes as well.
If someone calls `getWalletPublicKey` right after a fresh wallet creation there is a chance that chain endpoint hasn't synced yet and the retry would be desirable here.

Maybe we should add an optional argument `skipRetryWhenNotRegistered = false` and use this variant only in the redemption process?